### ISSUE TITLE
An NTRU back-end for HElib

### ISSUE THREAD

**Issue by shaih** [homenc/HElib#39]
HElib is probably mature enough now to start experimenting with other back-ends (i.e., other HE schemes), at least the ones that have the same plaintext spaces as BGV. In particular it would be nice to implement the homomorphic-NTRU scheme from [LTV, STOC 2012], see http://eprint.iacr.org/2013/094 (even without the multi-key option).

Several implementations of this HE schemes were reported, e.g., http://eprint.iacr.org/2014/389 and 
http://www.kaynar-rohloff.com/krohloff/papers/2014/Rohloff_Cousins_WAHC_2014_final.pdf. Perhaps they can be integrated with HElib, or at least serve as a starting point for implementing an NTRU back-end for HElib.

**Comment by bogdan-kulynych** [comment]
Correct me if I'm wrong, this means adding an alternative implementation of the crypto functionality in Ctxt.{h, cpp}?

**Comment by shaih** [comment]
Yes, and probably also FHE.{h, cpp}. We are currently looking into improving the APIs of this library, so it would make sense to wait with more back-ends until we decide what to do there.
